packages/legacy-compat/src/legacy-network-handler/legacy-network-handler.ts

```diff
@@ -231,7 +231,7 @@ function adapterDidInvalidate(
     if (serializer && typeof serializer.extractErrors === 'function') {
       let errorsHash = serializer.extractErrors(store, store.modelFor(identifier.type), error, identifier.id) as Record<
         string,
-        string | string[]
+        string | string[] | JsonApiError[]
       >;
       error.errors = errorsHashToArray(errorsHash);
```

Comment:
our extractErrors hook now returns JsonApiError[] but is still backwards compatible with hooks that users might have defined that return `string | string[]`